dashboard/app/models/user.rb

```diff
@@ -2764,6 +2766,26 @@ def should_validate_us_state?
     new_record? || us_state_changed?
   end
 
+  def student_in_lockout_flow?
```

Comment:
I would make the validation function private since you do not intend it to be called on its own

Reply:
It should also not be a `?` function since that implies it returns a boolean. I think the name should be updated to reflect its purpose... I recommended above something like `enforce_age_or_state_update`